app/jobs/whats_app_adapter/three_sixty_dialog/create_welcome_message_template_job.rb

```diff
@@ -5,14 +5,41 @@
 module WhatsAppAdapter
   module ThreeSixtyDialog
     class CreateWelcomeMessageTemplateJob < ApplicationJob
-      def perform(organization_id:, template_text:, token:)
+      def perform(organization_id:)
         @organization = Organization.find_by(id: organization_id)
-        @token = token
-        @template_text = template_text
+        @token = WhatsAppAdapter::ThreeSixtyDialog::TokenFetcherService.new.call
 
-        base_uri = ENV.fetch('THREE_SIXTY_DIALOG_PARTNER_REST_API_ENDPOINT', 'https://stoplight.io/mocks/360dialog/360dialog-partner-api/24588693')
-        partner_id = ENV.fetch('THREE_SIXTY_DIALOG_PARTNER_ID', nil)
-        waba_account_id = organization.three_sixty_dialog_client_waba_account_id
+        @base_uri = ENV.fetch('THREE_SIXTY_DIALOG_PARTNER_REST_API_ENDPOINT', 'https://stoplight.io/mocks/360dialog/360dialog-partner-api/24588693')
+        @partner_id = ENV.fetch('THREE_SIXTY_DIALOG_PARTNER_ID', nil)
+        @waba_account_id = organization.three_sixty_dialog_client_waba_account_id
+
+        if "welcome_message_#{organization.project_name.parameterize.underscore}".in?(existing_templates)
+          User.admin.find_each do |admin|
+            PostmarkAdapter::Outbound.welcome_message_updated!(admin, organization)
+          end
+        else
+          create_welcome_message_template
+        end
+      end
```

Comment:
There is a lot going on here. In a perfect world I would write sth. like this:
```ruby
def update_welcome_message_templates(organization)
  User.admin.find_each { |admin| PostmarkAdapter::Outbound.welcome_message_updated!(admin, organization) }
end

# ...

        template_name = "welcome_message_#{organization.project_name.parameterize.underscore}"
        if template_name.in?(existing_templates)
          update_welcome_message_templates(organization)
        else
          create_welcome_message_template
        end
      end
```

Reply:
sure, I like that the method explains what is happening.